staging/src/k8s.io/client-go/tools/cache/controller.go

```diff
@@ -121,20 +129,20 @@ func New(c *Config) Controller {
 	return ctlr
 }
 
-// Run begins processing items, and will continue until a value is sent down stopCh or it is closed.
-// It's an error to call Run more than once.
-// Run blocks; call via go.
-func (c *controller) Run(stopCh <-chan struct{}) {
+// RunWithStopOptions begins processing items, and will continue until stopOptions recognizes a stop condition.
+// RunWithStopOptions blocks; call via go.
+func (c *controller) RunWithStopOptions(stopOptions StopOptions) {
 	defer utilruntime.HandleCrash()
 	go func() {
-		<-stopCh
+		<-c.config.StopHandle.Done()
```

Comment:
What happens if `c.config` or `c.config.StopHandle` are nil?

Reply:
As mentioned in `reflector.go` I added new reflector constructors to hopefully alleviate some of the issues. 

While it doesn't solve the problem entirely, my understanding now is that at least we're not making it any worse (i.e. we would still run into the same panics regardless of this change if `c.config` is nil and there's nothing inherently more concerning about `c.config.StopHandle` being nil than there is if `c.config.Queue` is nil. 

Let me know what you think.